Include groups claim steps to Okta SSO page

akshay196 opened the pull request:
Based on this Paralus new feature: https://github.com/paralus/paralus/pull/172

akshay196 commented:
Showing many spell check errors now. :( Typos CI is not able to parse files correctly, it is purposefully built for source code.
@techmaharaj Do you know any alternative CI app for typos?

niravparikh05 commented:
@akshay196 let's include some more details like how to create a group, add people to groups and also a note that the group name should be the same as the one created in paralus having appropriate permissions

akshay196 commented:
> let's include some more details like how to create a group, add people to groups 

Adding guide about adding people to groups is not in scope of this page/Paralus documentation. Added a reference to Okta user/group management document. Hope it helps.

> and also a note that the group name should be the same as the one created in paralus having appropriate permissions

Good point. Added a note at the end.

techmaharaj commented:
@akshay196 I think it's important to mention why are we creating a claim here. Currently if you see the documentation, it just starts with 'Creating a Okta SSO Application` and then we talk about adding claims. There has to be a header somewhere that talks about what are claims and why we are adding one. 

Also, I don't see where have we mentioned that Okta group name must be similar to Paralus.

akshay196 commented:
> @akshay196 I think it's important to mention why are we creating a claim here. Currently if you see the documentation, it just starts with 'Creating a Okta SSO Application` and then we talk about adding claims. There has to be a header somewhere that talks about what are claims and why we are adding one.

ah, let me add more details.

> Also, I don't see where have we mentioned that Okta group name must be similar to Paralus.

Added a note at the end that goes as `Okta groups of the user are mapped to Paralus groups based on their names. The groups mapping provides necessary permissions to access Paralus resources after successful login. If a user is not part of any Okta group then admin needs to explicitly configure group for the user in Paralus.`.
Is it not clear?

techmaharaj commented:
> Added a note at the end that goes as `Okta groups of the user are mapped to Paralus groups based on their names. The groups mapping provides necessary permissions to access Paralus resources after successful login. If a user is not part of any Okta group then admin needs to explicitly configure group for the user in Paralus.`.
> Is it not clear? any suggestions?

Just saw it now, my bad.

I think instead of this being at the end of the document, it will make more sense to put it where you're adding the claim. 

So that's where you can mention two things:
- Why are we adding a claim, what it does and why it's required for Paralus
- This note that the group name should be the same as Paralus _(the user needs to create a group in Paralus first)_ and why.

akshay196 commented:
@techmaharaj @niravparikh05 Made all the changes. Please take a look.

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -17,21 +18,57 @@ Login to your Okta account as an admin and navigate to `Applications -> Applicat
+### Add a custom claim for groups mapping
+
+Paralus comes with the in-built functionality for mapping Identity
+provider (IdP) groups to Paralus. This is useful to manage different
```
`This is useful to manage different Paralus access to the IdP users through IdP groups they are part of. It eases the access-management process for Parlaus administrators`

We can rephrase this to "_This makes access management simpler. As an administrator, you don't need to manage access for an individual user but for a group of users_"

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -17,21 +18,57 @@ Login to your Okta account as an admin and navigate to `Applications -> Applicat
+provider (IdP) groups to Paralus. This is useful to manage different
+Paralus access to the IdP users through IdP groups they are part
+of. It eases the access-management process for Parlaus
+administrators. See the limitation below.
```
`See limitation below` - can be rephrased to 

_This however has a limitation. To map the groups correctly, the group names in Okta and Paralus must be the same. This way, the IdP users belonging to a group can access Paralus resources based on the permissions assigned to the group in Paralus. For example, you create a group called `DevTeam` in Paralus and the same group in Okta as well. Assign permissions to the Paralus group and add users to the Okta group. Now, when an IdP user belonging to the `DevTeam` group logs into Paralus using Okta, they will inherit the permissions set for the group. Read how to [manage users and groups in Okta](https://help.okta.com/oie/en-us/Content/Topics/users-groups-profiles/usgp-groups-main.htm)_

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -17,21 +18,57 @@ Login to your Okta account as an admin and navigate to `Applications -> Applicat
+[Okta
+documentation](https://help.okta.com/oie/en-us/Content/Topics/users-groups-profiles/usgp-groups-main.htm).
+
+Okta by default does not return groups information in OIdC
```
```suggestion
Okta by default does not return groups information in OIDC
```

Also, can we link to the Okta document that mentions the scopes so the reader can see that groups are not returned? I found this link: https://auth0.com/docs/get-started/apis/scopes/openid-connect-scopes

akshay196 replied:
This is generic document about OIDC scopes and standard claims. I am not able to find a document from Okta.

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -17,21 +18,57 @@ Login to your Okta account as an admin and navigate to `Applications -> Applicat
+
+<img src="/img/docs/oidc-okta-2.png" alt="Added groups claim to Okta authorization server" height="75%" width="75%"/>
+
+> **Limitation:** Currently after the IdP user logged in to Paralus,
```
```suggestion
> **Limitation:** Currently after the IdP user is logged in to Paralus,
```

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -17,21 +18,57 @@ Login to your Okta account as an admin and navigate to `Applications -> Applicat
+<img src="/img/docs/oidc-okta-2.png" alt="Added groups claim to Okta authorization server" height="75%" width="75%"/>
+
+> **Limitation:** Currently after the IdP user logged in to Paralus,
+> any change in Okta groups does not reflect on Paralus. This is a
```
```suggestion
> any change in Okta groups does not reflect in Paralus. This is a
```

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -17,21 +18,57 @@ Login to your Okta account as an admin and navigate to `Applications -> Applicat
+
+> **Limitation:** Currently after the IdP user logged in to Paralus,
+> any change in Okta groups does not reflect on Paralus. This is a
+> known gap. It is being tracked
```
```suggestion
> known gap. 
```

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -17,21 +18,57 @@ Login to your Okta account as an admin and navigate to `Applications -> Applicat
+> **Limitation:** Currently after the IdP user logged in to Paralus,
+> any change in Okta groups does not reflect on Paralus. This is a
+> known gap. It is being tracked
+> [here](https://github.com/paralus/paralus/issues/2).
```
```suggestion
> This is being tracked as a part of this [issue](https://github.com/paralus/paralus/issues/2).
```

techmaharaj commented on docs/Single Sign On/okta.md:
```diff
@@ -43,9 +80,7 @@ Click Save & Continue.
-The **Mapper Configuration** will be set for Okta, so you don't need to change anything on this screen.
-
-On the **Mapper Configuration** screen, provide `https://raw.githubusercontent.com/paralus/paralus/main/_kratos/oidc-mappers/okta.jsonnet` as the mapper url.
+On the **Mapper Configuration** screen, provide `https://raw.githubusercontent.com/paralus/paralus/main/_kratos/oidc-mappers/okta.jsonnet` as the mapper url. The mapper url has used [the custom claim added earlier for groups](#add-a-custom-claim-for-groups-mapping).
```
```suggestion
On the **Mapper Configuration** screen, provide `https://raw.githubusercontent.com/paralus/paralus/main/_kratos/oidc-mappers/okta.jsonnet` as the mapper url. The mapper url uses [the custom claim added earlier for groups](#add-a-custom-claim-for-groups-mapping).
```

techmaharaj reviewed (approved):
Left a few minor suggestions.

akshay196 commented:
@techmaharaj Updated PR. Please take a look.

akshay196 commented:
The spell checker workflow is reporting many false positives. Maintaining ignore list is an extra burden. I have removed the spell check in this PR. Hope this is fine.
